bigbluebutton-html5/imports/ui/services/user/mapUser.js

```diff
@@ -22,7 +22,7 @@ const mapUser = (user) => {
     isPresenter: user.presenter,
     isModerator: user.role === ROLE_MODERATOR,
     isCurrent: user.userId === userId,
-    isVoiceUser: !!voiceUser,
+    isVoiceUser: voiceUser.joined,
```

Comment:
Here is where you should add the business logic, not on the component.

`isTalking: voiceUser ? voiceUser.talking && !voiceUser.muted. : false,`